client/src/components/WorkspacePanels/MainWorkspacePanel/MainWorkspace.js

```diff
@@ -1,189 +1,89 @@
-        </Droppable>
-        <Droppable droppableId="droppable">
-          {(provided, snapshot) => (
-            <div
-              ref={provided.innerRef}
-              style={getListStyle(snapshot.isDraggingOver)}
-            >
-              {this.state.items.map((item, index) => (
-                <Draggable key={item.id} draggableId={item.id} index={index}>
-                  {(provided, snapshot) => (
-                    <div
-                      ref={provided.innerRef}
-                      {...provided.draggableProps}
-                      {...provided.dragHandleProps}
-                      style={getItemStyle(
-                        snapshot.isDragging,
-                        provided.draggableProps.style
-                      )}
-                    >
-                      {item.content}
-                    </div>
-                  )}
-                </Draggable>
+            {provided.placeholder}
+          </div>
+        )}
+      </Droppable>
+      <Droppable droppableId="timeline-drop-row-2">
+        {(provided, snapshot) => (
+          <div
+            ref={provided.innerRef}
+            style={getListStyle(snapshot.isDraggingOver)}
+          >
+            {music[1].length !== 0 &&
+              music[1].map(({ id, file, name, type }) => (
+                <div className="track-object" key={id}>
+                  <div className="type-display">{type}</div>
+                  <div className="track-data">{name}</div>
+                </div>
               ))}
-              {provided.placeholder}
-            </div>
-          )}
-        </Droppable>
-      </DragDropContext>
-    );
-  }
-}
+            {provided.placeholder}
+          </div>
+        )}
+      </Droppable>
+      <Droppable droppableId="timeline-drop-row-3">
+        {(provided, snapshot) => (
+          <div
+            ref={provided.innerRef}
+            style={getListStyle(snapshot.isDraggingOver)}
+          >
+            {music[2].length !== 0 &&
+              music[2].map(({ id, file, name, type }) => (
+                <div className="track-object" key={id}>
+                  <div className="type-display">{type}</div>
+                  <div className="track-data">{name}</div>
+                </div>
+              ))}
+            {provided.placeholder}
+          </div>
+        )}
+      </Droppable>
+      <Droppable droppableId="timeline-drop-row-4">
+        {(provided, snapshot) => (
+          <div
+            ref={provided.innerRef}
+            style={getListStyle(snapshot.isDraggingOver)}
+          >
+            {music[3].length !== 0 &&
+              music[3].map(({ id, file, name, type }) => (
+                <div className="track-object" key={id}>
+                  <div className="type-display">{type}</div>
+                  <div className="track-data">{name}</div>
+                </div>
+              ))}
+            {provided.placeholder}
+          </div>
+        )}
+      </Droppable>
```

Comment:
Do we need to hard-code this `Droppable` component multiple times according to the row num?

Reply:
good idea! i'll fix this